src/test/scala/com/microsoft/azure/iot/iothubreact/checkpointing/ConfigurationTest.scala

```diff
@@ -0,0 +1,57 @@
+package com.microsoft.azure.iot.iothubreact.checkpointing
+
+import com.microsoft.azure.iot.iothubreact.checkpointing.backends.cassandra.lib.Auth
+import com.typesafe.config.{Config, ConfigException}
+import org.scalatest.mockito.MockitoSugar
+import org.mockito.Mockito._
+import org.scalatest.{FeatureSpec, FunSuite, GivenWhenThen}
+
+class ConfigurationTest extends FeatureSpec with GivenWhenThen with MockitoSugar {
+
+  info("As a configured instance")
+  info("I want logic around returned values to be consistent with application expectations")
+
+  val confPath = "iothub-react.checkpointing."
+  feature("Configuration Cassandra authorization") {
+
+    scenario("Only one of username or password is supplied") {
+      var cfg = mock[Config]
+      when(cfg.getString(confPath + "storage.cassandra.username")).thenReturn("username")
+      when(cfg.getString(confPath + "storage.cassandra.password")).thenThrow(new ConfigException.Missing("path"))
+      assert(new Configuration(cfg).cassandraAuth == None)
+
+      cfg = mock[Config]
+      when(cfg.getString(confPath + "storage.cassandra.username")).thenThrow(new ConfigException.Missing("path"))
+      when(cfg.getString(confPath + "storage.cassandra.password")).thenReturn("password")
+      assert(new Configuration(cfg).cassandraAuth == None)
+    }
+
+    scenario("Both username and password are supplied") {
+      var cfg = mock[Config]
+      when(cfg.getString(confPath + "storage.cassandra.username")).thenReturn("username")
+      when(cfg.getString(confPath + "storage.cassandra.password")).thenReturn("password")
+      assert(new Configuration(cfg).cassandraAuth == Some(Auth("username","password")))
```

Comment:
please check the code style (I recommend Save Actions and enable "autoformatting" using the IntelliJ codestyle we've shared). (in this line there should be a space after `"username",`